background.js

```diff
@@ -0,0 +1,161 @@
+// x: the origin point of the x (horizontal) cardinal direction for drawing
+// y: the origin point of the y (vertical) cardinal direction for drawing
+class Grasscorner {
+    constructor(game, x, y) {
+        Object.assign(this, { game, x, y });
+
+        this.spritesheet = ASSET_MANAGER.getAsset("./sprites/ground_sprites.png");
+    };
+
+    update() {
+
+    };
+
+    // drawImage(spritesheet, sx, sy, sw, sh, dx, dy, dw, dh)
+    draw(ctx) {
+        //top left corner
+        ctx.drawImage(this.spritesheet, 0, 1216, params.TILE_W_H, params.TILE_W_H, 0, 0, params.TILE_W_H, params.TILE_W_H);
+        let yCanvasOrigin = params.CANVAS_HEIGHT - params.TILE_W_H;
+        let xCanvasOrigin = params.CANVAS_WIDTH - params.TILE_W_H;
+        //bottom left corner
+
+        ctx.save();
+        ctx.scale(1,-1);
+        ctx.drawImage(this.spritesheet, 0, 1216, params.TILE_W_H, params.TILE_W_H, 0, -yCanvasOrigin - params.TILE_W_H, params.TILE_W_H, params.TILE_W_H);
+        ctx.restore();
+
+        //top right corner
+        ctx.save();
+        ctx.scale(-1,1);
+        ctx.drawImage(this.spritesheet, 0, 1216, params.TILE_W_H, params.TILE_W_H, -xCanvasOrigin - params.TILE_W_H, 0, params.TILE_W_H, params.TILE_W_H);
+        ctx.restore();
+
+        //bottom right corner
+        ctx.save();
+        ctx.scale(-1,-1);
+        ctx.drawImage(this.spritesheet, 0, 1216, params.TILE_W_H, params.TILE_W_H, -xCanvasOrigin - params.TILE_W_H, -yCanvasOrigin - params.TILE_W_H, params.TILE_W_H, params.TILE_W_H);
+        ctx.restore();
+    };
+};
+
+// This class draws the along the walls of the canvas for the game map.
+// game: the current iteration of the GameEngine
+// x: the origin point of the x (horizontal) cardinal direction for drawing
+// y: the origin point of the y (vertical) cardinal direction for drawing
+class Vertwall {
+    constructor(game, x, y) {
+        Object.assign(this, { game, x, y });
+
+        this.spritesheet = ASSET_MANAGER.getAsset("./sprites/ground_sprites.png");
+    };
+
+    update() {
+
+    };
+
+    // drawImage(spritesheet, sx, sy, sw, sh, dx, dy, dw, dh)
+    draw(ctx) {
+      for (var i = 0; i < params.VERT_WALL_COUNT; i++) {
+          ctx.drawImage(this.spritesheet, 0, 1247, params.TILE_W_H, params.TILE_W_H, this.x, this.y + i * params.TILE_W_H, params.TILE_W_H, params.TILE_W_H);
+
+          ctx.save();
+          ctx.scale(-1,1);
+          // The dx here is a copy of the one in Grasscorner above.
+          // -(this.x + (params.CANVAS_WIDTH - params.TILE_W_H)) = -(0 + (1024 - 64)) - 64
+          // First it flips the original image and puts it off the canvas. The -64 shifts it left to be visible once more.
+          ctx.drawImage(this.spritesheet, 0, 1247, params.TILE_W_H, params.TILE_W_H, -(this.x + (params.CANVAS_WIDTH - params.TILE_W_H)) - params.TILE_W_H,
+                        this.y + i * params.TILE_W_H, params.TILE_W_H, params.TILE_W_H);
+          ctx.restore();
+      }
+    };
+};
```

Comment:
nice work!: not sure I quite understand the numbers behind this. but I think I understand the gist of it.